File: 2195-time-needed-to-buy-tickets/time-needed-to-buy-tickets.java
class Solution {
    public int timeRequiredToBuy(int[] tickets, int k) {
        
        int time = tickets[k];

        int kth = tickets[k];

        for(int i = 0 ; i < k ; i++){
            if(tickets[i] > kth){
                time+=kth;
            }
            else{
                time+=tickets[i];
            }
        }

        for(int i=k+1 ; i<tickets.length ; i++){
            
            if(tickets[i] > kth-1 ){
                time+=kth-1;
            }
            else{
                time+=tickets[i];
            }

        }

        return time;

    }
}

File: 2195-time-needed-to-buy-tickets/time-needed-to-buy-tickets.py
from collections import deque

class Solution:
    def timeRequiredToBuy(self, tickets: list[int], k: int) -> int:

        q = deque()

        for i in range(len(tickets)):
            q.append(i)
        
        time = 0

        while(tickets[k] != 0):
            top = q[0]
            tickets[top]-=1
            q.popleft()
            time+=1
            if(tickets[top] > 0):
                q.append(top)
        
        return time